src/main/java/io/antmedia/AntMediaApplicationAdapter.java

```diff
@@ -1173,6 +1176,65 @@ private void putToMap(String keyName, Object keyValue, Map<String, Object> map)
 		}
 	}
 
+
+	public boolean sendClusterPost(String url, String clusterCommunicationToken, int retryAttempts) {
+		logger.info("Sending cluster POST request to {}", url);
+		try (CloseableHttpClient httpClient = getHttpClient()) {
+			HttpPost httpPost = new HttpPost(url);
+			RequestConfig requestConfig = RequestConfig.custom()
+					.setConnectTimeout(CLUSTER_POST_TIMEOUT_MS)
+					.setConnectionRequestTimeout(CLUSTER_POST_TIMEOUT_MS)
+					.setSocketTimeout(CLUSTER_POST_TIMEOUT_MS)
+					.build();
+			httpPost.setConfig(requestConfig);
+
+			httpPost.setHeader(TokenFilterManager.TOKEN_HEADER_FOR_NODE_COMMUNICATION, clusterCommunicationToken);
+
+			try (CloseableHttpResponse httpResponse = httpClient.execute(httpPost)) {
+				int statusCode = httpResponse.getStatusLine().getStatusCode();
+				logger.info("Cluster POST Response Status: {}", statusCode);
+				if (statusCode != HttpStatus.SC_OK) {
+					if (retryAttempts >= 1) {
+						logger.info("Retry attempt for Cluster POST in {} milliseconds due to non-200 response: {}",
+								appSettings.getWebhookRetryDelay(), statusCode);
+						return retrySendClusterPostWithDelay(url, clusterCommunicationToken, retryAttempts - 1);
+					} else {
+						logger.info("Stopping sending Cluster POST because no more retry attempts left. Giving up.");
+						return false;
+					}
+				}
+				return true;
+			}
+		} catch (IOException e) {
+			if (retryAttempts >= 1) {
+				logger.info("Retry attempt for Cluster POST in {} milliseconds due to IO exception: {}",
+						appSettings.getWebhookRetryDelay(), ExceptionUtils.getStackTrace(e));
+				return retrySendClusterPostWithDelay(url, clusterCommunicationToken, retryAttempts - 1);
+			} else {
+				logger.info("Stopping sending Cluster POST because no more retry attempts left. Giving up.");
+				return false;
+			}
+		}
+	}
+
+	public boolean retrySendClusterPostWithDelay(String url, String clusterCommunicationToken, int retryAttempts) {
+		CompletableFuture<Boolean> future = new CompletableFuture<>();
+
+		vertx.setTimer(appSettings.getWebhookRetryDelay(), timerId -> {
+			boolean result = sendClusterPost(url, clusterCommunicationToken, retryAttempts);
+			future.complete(result);
+		});
+
+		try {
+			// Timeout for each retry = retry delay + request timeout + buffer
+			return future.get(appSettings.getWebhookRetryDelay() + CLUSTER_POST_TIMEOUT_MS + 500, TimeUnit.MILLISECONDS);
```

Comment:
It's a blocking call. I mean it waits until the operation has finished. Please don't call blocking calls. Make everything running all the time.  

I understand that you need to return a response to the method and it is also being called by REST method.

My suggestion is 
- Seturn the `Result` as `success:true` 
- Set an `errorId` in `Result` it will make developers understand there is a relay in the stream
- Set  extra information into `message` field by saying something like ->  this request is being forwarded to the origin to be broadcasted. To make sure it's started please get broadcast and check its status.